flujo/architect/states/finalization.py

```
@@ -0,0 +1,145 @@
+                        "agent": {"id": "flujo.builtins.stringify", "params": {}},
+                    }
+                    block = _yaml.safe_dump(step_dict, sort_keys=False).strip()
+                    steps_block = "\n".join(
+                        [
+                            "- " + line if i == 0 else "  " + line
+                            for i, line in enumerate(block.splitlines())
+                        ]
+                    )
+                    yaml_text = f'\nversion: "0.1"\nname: {name}\nsteps:\n{steps_block}\n'
+                    try:
+                        setattr(context, "generated_yaml", yaml_text)
```

Ruff (0.14.5)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)